source/docs/software/basic-programming/coordinate-system.rst

```diff
@@ -0,0 +1,325 @@
+
+.. figure:: images/coordinate-system/rotation.svg
+   :scale: 200
+   :alt: Unit circle
+
+   Unit circle with common angles
+
+The figure above shows the unit circle with common angles labeled in degrees (°) and radians (rad). Notice that rotation to the right is negative, and the range for the whole unit circle is -180° to 180° (-Pi radians to Pi radians).
+
+.. note:: The range is (-180, 180], meaning it is exclusive of -180° and inclusive of 180°.
+
+There are some places you may choose to use a different range, such as 0° to 360° or 0 to 1 rotation, but be aware that many core WPILib classes and FRC tools are built with the unit circle above.
+
+.. warning:: Some gyros/IMUs use CW positive rotation, such as the NavX IMU. Care must be taken to handle rotation properly, sensor values may need to be inverted. Read the documentation and verify that rotation is CCW positive.
+
+.. warning:: Many sensors that read rotation around an axis, such as encoders and IMU's, read continuously. This means they read more than one rotation, so when rotating past 180° they read 181°, not -179°. Some sensors have configuration settings where you can choose their wrapping behavior and range, while others need to be handled in your code. Careful attention should be paid to make sure sensor readings are consistent and your control loop handles wrapping in the same way as your sensor.
+
+Joystick and controller coordinate system
+-----------------------------------------
+
+Joysticks, including the sticks on controllers, don't use the same NWU coordinate system. They use the NED (North-East-Down) convention, where the positive X axis points ahead, the positive Y axis points right, and the positive Z axis points down. When viewed with each positive axis pointing toward you, counter-clockwise (CCW) is a positive value and clockwise (CW) is a negative value.
+
+.. figure:: images/coordinate-system/joystick-3d.svg
+   :scale: 200
+   :alt: NED axes convention
+
+   Joystick coordinate system
+
+It's important to note that joystick axes values are rotations around the respective axes, not translations. In practical terms, this means:
+
+- pushing forward on the joystick (toward the positive X axis) is a CW rotation around the Y axis, so you get a negative Y value.
+- pushing to the right (toward the postivie Y axis) is a CCW rotation around the X axis, so you get a positive X value.
+- twisting the joystick CW (toward the positive Y axis) is a CCW rotation around the Z axis, so you get a positive Z value.
+
+Using Joystick and controller input to drive a robot
+---------------------------------------------------------
+
+You may have noticed, the coordinate system used by WPILib for the robot is not the same as the coordinate system used for joysticks and controllers. Care needs to be taken to understand the difference, and properly pass driver input to the drive subsystem.
+
+Non-holonomic drivetrain example
```

Comment:
I think it would be more keeping with the rest of the docs to say Differential drivetrain, rather then non-holonomic, especially since that is the only non-holonomic drivetrain that WPILib supports

Reply:
I agree. I did that heading when I wasn't sure where I was going to put swerve. I'll update it.